pkg/onboarding/usecases/shortcode.go

```diff
@@ -0,0 +1,155 @@
+		baseExt:              ext,
+		engagement:           engage,
+		pubSub:               ps,
+		hubspotCRM:           crm,
+	}
+}
+
+// CreateSMSData saves sent message, replies to the sender and creates hubspot contact
+func (s *SMSImpl) CreateSMSData(ctx context.Context, input *dto.AfricasTalkingMessage) error {
+	ctx, span := tracer.Start(ctx, "CreateSMSData")
+	defer span.End()
+
+	validatedInput, err := utils.ValidateAficasTalkingSMSData(input)
+	if err != nil {
+		utils.RecordSpanError(span, err)
+		return err
+	}
+
+	// Checking if the sender has a profile
+	profile, err := s.onboardingRepository.GetUserProfileByPhoneNumber(ctx, validatedInput.From, false)
+	if err != nil {
+		utils.RecordSpanError(span, err)
+		//should not panic when the user profile is not found
+		log.Errorf("an error occurred: %v", err)
+	}
+
+	if profile != nil {
+		// Replying to the customer without the download link and notifying the admin
+		to := validatedInput.From
+		message := bewellUserReplyMessage
+
+		supportEmailPayload := &dto.EmailNotificationPayload{
+			SubjectTitle: validatedInput.Text,
+			EmailBody:    validatedInput.Text,
+			PrimaryPhone: validatedInput.From,
+			BeWellUser:   domain.GeneralOptionTypeYes,
+			Time:         validatedInput.Date,
+		}
+
+		return s.replyAndNotifyAdmin(ctx, to, message, validatedInput, supportEmailPayload)
+	}
+
+	//Reply back with the download link for non-bewell user(new user)
+	to := validatedInput.From
+	message := newSenderReplyMessage
+
+	supportEmailPayload := &dto.EmailNotificationPayload{
+		SubjectTitle: validatedInput.Text,
+		EmailBody:    validatedInput.Text,
+		PrimaryPhone: validatedInput.From,
+		BeWellUser:   domain.GeneralOptionTypeNo,
+		Time:         validatedInput.Date,
+	}
+
+	err = s.replyAndNotifyAdmin(ctx, to, message, validatedInput, supportEmailPayload)
+	if err != nil {
+		return err
+	}
+
+	//Create CRM contact for the new user
+	contact := &domain.CRMContact{
+		Properties: domain.ContactProperties{
+			Phone:                 validatedInput.From,
+			Gender:                string(domain.GeneralOptionTypeNotGiven),
+			FirstChannelOfContact: domain.ChannelOfContactShortcode,
+			BeWellEnrolled:        domain.GeneralOptionTypeNo,
+			OptOut:                domain.GeneralOptionTypeNo,
+		},
+	}
+
+	err = s.pubSub.NotifyCreateContact(ctx, *contact)
+	if err != nil {
+		utils.RecordSpanError(span, err)
+		log.Printf("failed to publish to crm.contact.create topic: %v", err)
+	}
+
+	return nil
+}
+
+func (s *SMSImpl) replyAndNotifyAdmin(
+	ctx context.Context,
+	to string,
+	message string,
+	validatedInput *dto.AfricasTalkingMessage,
+	supportEmailPayload *dto.EmailNotificationPayload) error {
+	ctx, span := tracer.Start(ctx, "CreateSMSData")
+	defer span.End()
+
+	err := s.engagement.SendSMS(ctx, []string{to}, message)
+	if err != nil {
+		return fmt.Errorf("an error occurred while sending SMS: %v", err)
+	}
+
+	err = s.engagement.NotifyAdmins(ctx, *supportEmailPayload)
```

Comment:
Attach a screenshot of the email being sent, I'd like to review that too.

Reply:
![Screenshot from 2021-08-23 10-56-15](https://user-images.githubusercontent.com/41376826/130411424-633b1ea7-554a-489d-9b94-6d16868642cb.png)

Reply:
![Screenshot from 2021-08-23 15-24-45](https://user-images.githubusercontent.com/41376826/130582285-e373db9d-e3e4-4340-981d-c010d78fd385.png)

Reply:
@wshikami Kindly assist with the email copy